src/main/java/com/yelp/nrtsearch/server/luceneserver/RegisterFieldsHandler.java

```diff
@@ -132,6 +133,27 @@ public FieldDefResponse handle(final IndexState indexState, FieldDefRequest fiel
         indexState.addField(ent.getValue(), fieldAsJsonObject);
       }
     }
+
+    if (!indexState.getAllFields().containsKey("_nested_path")) {
+      // register _nested_path field to every index to differentiate root
+      // and child documents
```

Comment:
I can only add a registered field. If _nested_path is not nested. I am not able to document add it.

Reply:
You are only adding the the lucene Document, right? This does not do any check against the nrtsearch schema.

Reply:
I got this exception
`Caused by: java.lang.IllegalArgumentException: field "_nested_path" is unknown: it was not registered with registerField
	at com.yelp.nrtsearch.server.luceneserver.IndexState.getField(IndexState.java:671)
	at com.yelp.nrtsearch.server.luceneserver.ServerCodec.getPostingsFormatForField(ServerCodec.java:43)
	at org.apache.lucene.codecs.lucene84.Lucene84Codec$1.getPostingsFormatForField(Lucene84Codec.java:67)
	at org.apache.lucene.codecs.perfield.PerFieldPostingsFormat$FieldsWriter.buildFieldsGroupMapping(PerFieldPostingsFormat.java:219)
	at org.apache.lucene.codecs.perfield.PerFieldPostingsFormat$FieldsWriter.write(PerFieldPostingsFormat.java:151)
	at org.apache.lucene.index.FreqProxTermsWriter.flush(FreqProxTermsWriter.java:97)
	at org.apache.lucene.index.DefaultIndexingChain.flush(DefaultIndexingChain.java:177)
	at org.apache.lucene.index.DocumentsWriterPerThread.flush(DocumentsWriterPerThread.java:468)
	at org.apache.lucene.index.DocumentsWriter.doFlush(DocumentsWriter.java:555)
	at org.apache.lucene.index.DocumentsWriter.flushAllThreads(DocumentsWriter.java:722)
	at org.apache.lucene.index.IndexWriter.getReader(IndexWriter.java:494)
	at org.apache.lucene.index.StandardDirectoryReader.doOpenFromWriter(StandardDirectoryReader.java:297)
	at org.apache.lucene.index.StandardDirectoryReader.doOpenIfChanged(StandardDirectoryReader.java:272)
	at org.apache.lucene.index.StandardDirectoryReader.doOpenIfChanged(StandardDirectoryReader.java:262)
	at org.apache.lucene.index.DirectoryReader.openIfChanged(DirectoryReader.java:165)
	at org.apache.lucene.facet.taxonomy.SearcherTaxonomyManager.refreshIfNeeded(SearcherTaxonomyManager.java:147)
	at org.apache.lucene.facet.taxonomy.SearcherTaxonomyManager.refreshIfNeeded(SearcherTaxonomyManager.java:41)
	at org.apache.lucene.search.ReferenceManager.doMaybeRefresh(ReferenceManager.java:176)
	at org.apache.lucene.search.ReferenceManager.maybeRefreshBlocking(ReferenceManager.java:253)
	at com.yelp.nrtsearch.server.luceneserver.ShardState.maybeRefreshBlocking(ShardState.java:982)
	at com.yelp.nrtsearch.server.grpc.LuceneServer$LuceneServerImpl.refresh(LuceneServer.java:672)
	at com.yelp.nrtsearch.server.grpc.LuceneServerGrpc$MethodHandlers.invoke(LuceneServerGrpc.java:2664)
	at io.grpc.stub.ServerCalls$UnaryServerCallHandler$UnaryServerCallListener.onHalfClose(ServerCalls.java:172)
	at io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.halfClosed(ServerCallImpl.java:331)
	at io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1HalfClosed.runInContext(ServerImpl.java:820)
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	... 1 more`

Reply:
I see, it needs the postings format. In that case, maybe add a static collection of meta FieldDefs to IndexState and change getField to check the meta fields first. This should keep it out of the schema, but still make it available.

You would also want to use this FieldDef to add fields to the document. i.e. metaFieldDef.parseDocumentField(...) instead of document.add(new StringField(...))